src/lib.rs

```diff
@@ -257,9 +269,15 @@ impl<T: AsRef<str>> Grid<T> {
 
 impl<T: AsRef<str>> fmt::Display for Grid<T> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
-        let separator = match &self.options.filling {
-            Filling::Spaces(n) => " ".repeat(*n),
-            Filling::Text(s) => s.clone(),
+        // If cells are empty then, nothing to print, skip.
+        if self.cells.is_empty() {
+            return Ok(());
+        }
+
+        let (tab_size, separator) = match &self.options.filling {
```

Comment:
I wonder whether using `Option<usize>` (or maybe `Option<NonZeroUsize>`) might be nicer here, because `0` has a special meaning here.

Reply:
If I got you right, you want to do something like:
```rust
let (tab_size, separator) = match &self.options.filling {
    Filling::Spaces(n) => (None, " ".repeat(*n)),
    Filling::Text(s) => (None, s.clone()),
    Filling::Tabs { spaces, tab_size } => (Some(*tab_size), " ".repeat(*spaces)),
};

// ...

match tab_size {
    Some(tab_size) => {}, // Calculate tabs.
    None => {}, // Print spaces.
}
```

If that's what you mean, I'll disagree in this design. With this change I can do something stupid as pass `Some(0)` as a `tab_size` and instead of simply filling everything with spaces, it will still try to calculate the tabs size.

Reply:
That is indeed what I would suggest, but the `0` tab size indeed should not be possible, so it would be:
```
Filling::Tabs { spaces, tabsize } => (NonZeroUsize::new(*tab_size), " ".repeat(*spaces)),
```
It would functionally be _very_ similar to what you have, but it would make it very clear that `None` is a very different case than positive numbers. Feel free to leave it like this though. It's probably clear enough.

See https://doc.rust-lang.org/stable/std/num/type.NonZeroUsize.html for more info on that type.

Reply:
Thx for the link, didn't think about that before. I tried this solution, but got an error in calculations of the tabs **cannot subtract `usize` from `NonZero<usize>`**.  Even tho I like the suggestion, but is it reasonable to add more types conversions to handle this update?

Reply:
Yes! You can just do a `get` after matching on the option to get a usize

Reply:
Yeah, I've read about it. But my question is mostly how justified is it, to add new type, just to replace *if-else* on this:
```rust
match tab_size {
    Some(tab_size) => {
        let tab_size = tab_size.get();
        // ...
        }
    None => {
        // ...
    }
}
```

Reply:
You could write that with an `if let` if you prefer:
```rust
if let Some(tab_size) = tab_size {

} else {

}
```
But I do see your point. It's up to you!